### ISSUE TITLE
UdpSocket::recv eats 100% cpu

### ISSUE THREAD

**Issue by khvzak** [async-rs/async-std#803]
The following example:
```rust
use async_std::net::UdpSocket;

#[async_std::main]
async fn main() -> std::io::Result<()> {
    let socket = UdpSocket::bind("127.0.0.1:8080").await?;
    println!("Listening on {}", socket.local_addr()?);

    let mut buf = vec![0u8; 1024];
    loop {
        let n = socket.recv(&mut buf).await?;
        println!("Received: {:?}", &buf[..n]);
    }
}
```

Eats 100% cpu on macOS & Linux when idle (rustc 1.43.1)

Affected version: 1.6
On the previous 1.5 works well.

**Comment by dignifiedquire** [comment]
@khvzak can you check what version of `smol` you were using. `smol@0.1.11` was just released which has some fixes that could help here.

**Comment by khvzak** [comment]
Same issue after running `cargo update` (`smol@0.1.11`).
300% CPU according `htop` on macOS

**Comment by dignifiedquire** [comment]
thanks :(

cc @stjepang

**Comment by stjepang** [comment]
This should be fixed in smol v0.1.12. Can you do `cargo update` and see if the issue is resolved?

**Comment by khvzak** [comment]
Works fine now, thanks!
